src/test/java/io/strimzi/kafka/bridge/mqtt/core/MqttKafkaMapperTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.bridge.mqtt.core;
+
+import io.strimzi.kafka.bridge.mqtt.utils.MappingRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertThrows;
+
+/**
+ * Unit tests for {@link MqttKafkaMapper}
+ */
+public class MqttKafkaMapperTest {
+
+
+    /**
+     * Test for default topic.
+     * If the MQTT topic does not match any of the mapping rules, the default topic is used.
+     * E.g. if the Topic Mapping Rules is empty, the default topic is used by default.
+     */
+    @Test
+    public void testDefaultTopic() {
+        List<MappingRule> rules = new ArrayList<>();
+
+        MqttKafkaMapper mapper = new MqttKafkaMapper(rules);
+
+        assertThat("Should use the default topic when no mapping pattern matches.",
+                mapper.map("sensor/temperature"), is(MqttKafkaMapper.DEFAULT_KAFKA_TOPIC));
+    }
+
+    /**
+     * Test the mapping of single level topics.
+     */
+    @Test
+    public void testSingleLevel() {
+        List<MappingRule> rules = new ArrayList<>();
+
+        rules.add(new MappingRule("sensor_data", "sensors/+/data"));
+        rules.add(new MappingRule("devices_{device}_data", "devices/{device}/data"));
+        rules.add(new MappingRule("fleet_{fleet}", "fleet/{fleet}/vehicle/{vehicle}"));
+
+        MqttKafkaMapper mapper = new MqttKafkaMapper(rules);
+
+        assertThat("Mqtt pattern sensors/+/data should be mapped to sensor_data",
+                mapper.map("sensors/4/data"), is("sensor_data"));
+
+        assertThat("Mqtt pattern devices/{device}/data should be mapped to devices_{device}_data",
+                mapper.map("devices/4/data"), is("devices_4_data"));
+
+        assertThat("Mqtt pattern fleet/{fleet}/vehicle/{vehicle} should be mapped to fleet_{fleet}",
+                mapper.map("fleet/4/vehicle/23"), is("fleet_4"));
+    }
+
+
+    /**
+     * Test the mapping of single level topics.
+     */
+    @Test
+    public void testIllegalPlaceholder(){
+
+        List<MappingRule> rules = new ArrayList<>();
+        rules.add(new MappingRule("fleet_{fleet}", "fleet/{flee}/vehicle/{vehicle}"));
+
+        MqttKafkaMapper mapper = new MqttKafkaMapper(rules);
+
+        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
+            mapper.map("fleet/4/vehicle/23");
+        });
```

Comment:
I am not sure this should return an exception. In my mind when a match is not found we should use the default topic. Or maybe better, it has to be something configurable by the user.

Reply:
Of course `fleet{number}` is not a good Kafka topic if the MQTT pattern is `fleet/#` but if the pattern is `fleet/{number}` and the user sends to `fleet/10`, then `fleet10` will be a valid Kafka topic.